subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

```diff
@@ -225,3 +225,44 @@ Now, it is officially annotated as deprecated and scheduled for removal in Gradl
 
 The convention properties contributed by the `java` plugin have been deprecated and scheduled for removal in Gradle 8.0.
 They are replaced by the the properties of link:{groovyDslPath}/org.gradle.api.plugins.JavaPluginExtension.html[JavaPluginExtension] which can we configured in the `java {}` block.
+
+[[code_quality_plugin_configuration_consumption]]
+=== Deprecated consumption of code quality plugin configurations
+
+The following code quality plugins each declare a configuration with the same name as the plugin:
+
+- `codenarc`
+- `pmd`
+- `checkstyle`
+
+Those configurations are used to declare dependencies for the respective tool's runtime classpath.
+They are not meant for consumption outside of the project applying the plugin, but Gradle did not explicitly
+prohibit this. Gradle 7.1 deprecates such consumption and it will become an error in Gradle 8.0.
+
+If your use case needs to consume the configurations of the above tools in another project, please create a separate consumable
+configuration that extends from the internal ones. For example:
+```
+plugins {
+    id("codenarc")
+}
+configurations {
+    codenarc {
+        // because currently this is consumable until Gradle 8.0 and can clash with the configuration below depending on the attributes set
+        canBeConsumed = false
```

Comment:
On the other hand, once we add attributes to this configuration and later make `canBeConsumed = false` the default in Gradle 8, merely setting `canBeConsumed = true` here can make it consumable again without the need to extend and declare a separate configuration. Not sure if this example is what we want to show in this case.

Or can we eventually make this strictly not consumable and disallow flipping this flag?